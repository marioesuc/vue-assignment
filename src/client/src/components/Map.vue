<template>
  <div id="map">
    <div id="mapShadow">
    </div>
    <l-map
      id="mapLayer"
      :zoom="zoom"
      :center="center"
      :options="mapOptions"
      style="height: 100%"
      @update:center="centerUpdate"
      @update:zoom="zoomUpdate"
    >
      <l-tile-layer :url="url" :attribution="attribution" />
      <l-marker :lat-lng="withPopup">
        <l-popup>
          <div @click="innerClick">
            I am a popup
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque
              sed pretium nisl, ut sagittis sapien. Sed vel sollicitudin nisi.
              Donec finibus semper metus id malesuada.
            </p>
          </div>
        </l-popup>
      </l-marker>
    </l-map>
  </div>
</template>

<script>
import { latLng } from "leaflet";
import { LMap, LTileLayer, LMarker, LPopup } from "vue2-leaflet";

export default {
  name: 'Map',
  components: {
    LMap,
    LTileLayer,
    LMarker,
    LPopup,
  },
  props: {
    latitude: Number,
    longitude: Number,
  },
  data() {
    return {
      zoom: 15,
      center: latLng(this.latitude, this.longitude),
      url: 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
      attribution:
        '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
      withPopup: latLng(this.latitude, this.longitude),
      currentZoom: 11.5,
      currentCenter: latLng(this.latitude, this.longitude),
      mapOptions: {
        zoomSnap: 0.5
      },
    };
  },
  watch: {
    latitude() {
      this.center = latLng(this.latitude, this.longitude)
      this.withPopup = latLng(this.latitude, this.longitude)
      this.currentCenter = latLng(this.latitude, this.longitude)
    }
  },
  methods: {
    zoomUpdate(zoom) {
      this.currentZoom = zoom;
    },
    centerUpdate(center) {
      this.currentCenter = center;
    },
    showLongText() {
      this.showParagraph = !this.showParagraph;
    },
    innerClick() {
      alert('Click!');
    }
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
#map {
  position: relative;
  width: 800px;
  flex: 1;
  border: 3px solid white;
}
#mapLayer {
  border-radius: 15px;
  box-shadow: inset 0px 0px 10px 0px rgba(0,0,0,0.5);
}
#mapShadow {
  position: absolute;
  width: 100%;
  height: 100%;
  border-radius: 15px;
  box-shadow: inset 0px 0px 10px 0px rgba(0,0,0,0.5);
  z-index: 999;
}
</style>
