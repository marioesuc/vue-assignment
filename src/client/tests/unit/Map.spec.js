import { shallowMount } from "@vue/test-utils";
import Map from "@/components/Map.vue";
import { LMap, LTileLayer, LMarker } from "vue2-leaflet";

describe("Map.vue", () => {
  it("renders correctly", () => {
    const propsData = {
      latitude: 52.093448638916016,
      longitude: 5.117378234863281
    };

    const wrapper = shallowMount(Map, { propsData });
    expect(wrapper.find(LMap).exists()).toBeTruthy();
    expect(wrapper.find(LTileLayer).exists()).toBeTruthy();
    expect(wrapper.find(LMarker).exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
