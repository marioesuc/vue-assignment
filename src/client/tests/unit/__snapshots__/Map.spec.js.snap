// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map.vue renders correctly 1`] = `
<div id="map">
  <l-map-stub options="[object Object]" center="LatLng(52.093449, 5.117378)" zoom="15" crs="[object Object]" id="mapLayer">
    <div id="mapShadow"></div>
    <l-tile-layer-stub pane="tilePane" attribution="&amp;copy; <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors" visible="true" opacity="1" zindex="1" tilesize="256" subdomains="abc" options="[object Object]" url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" tilelayerclass="function tileLayer(url, options) {
	return new TileLayer(url, options);
}"></l-tile-layer-stub>
    <l-marker-stub pane="markerPane" visible="true" options="[object Object]" latlng="LatLng(52.093449, 5.117378)" icon="[object Object]"></l-marker-stub>
  </l-map-stub>
</div>
`;
